### PR TITLE
Adding Rspec Features

### PR DESCRIPTION
**Pull request by jah2488**
I've still got a lot more to write, but I think its best to get these changes into the code base sooner than later. These aren't meant to be exhaustive and they suffer from [INSERT COMPLAINT OF INTEGRATION SPECS HERE] as expected, but they're relatively fast and will help us get our testing harness up so we don't accidentally break a feature.

These specs focus on picking a cohort and making sure all the appropriate data is rendered (and clickable) on the instructor 'dashboard' which I guess now is the 'cohort show' page. 

Next I'll be adding some more scenarios for students using their dashboard.

Then last will be login/signup and reports. That'll give us pretty good high level coverage of 90% of the app. Will certainly help improve our confidence when merging pull requests and deploying.

### CODE REVIEW

**Comment by kurtisnelson** [comment]
What's your view on integration vs view?

**Comment by jah2488** [comment]
@kurtisnelson Honestly, I've found view specs rather pointless unless I'm trying to maintain the UI of some interface exactly and needs dozens and dozens of tests that need to run quickly.

I tend to favor a broad set of full integration tests, not exhaustive, but enough to cover the major features and then lots of model tests. In an ideal world, all controllers would be wafer thin and there would be no controller tests either, but I'll leave that as a goal for now.

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ def react_md(text, props = {}, opts = {})
   end
 
   def markdown(source)
-    Kramdown::Document.new(source).to_html.html_safe
+    Kramdown::Document.new(source || '').to_html.html_safe
```
Comment (jah2488) [inline app/helpers/application_helper.rb]: I am honestly surprised that this one hasn't bit me yet, but I'm glad I caught it. Yay tests!

File: spec/models/assignment_spec.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@
         am_three.due_date.beginning_of_week.to_date => [ am_three ],
         am_four.due_date.beginning_of_week.to_date  => [ am_four ]
       }
-      expect(Assignment.by_week(records)).to include(result)
+      expect(Assignment.by_week(records)).to eq result
```
Comment (jah2488) [inline spec/models/assignment_spec.rb]: This should fix that heisen-bug

**Comment by jah2488** [comment]
@kurtisnelson I also add I approve of any tests over no tests :+1:

**Pull request merged**